common/djangoapps/student/models_api.py

```diff
@@ -46,3 +47,11 @@ def create_manual_enrollment_audit(
 
 def get_course_enrollment(user, course_run_key):
     return _CourseEnrollment.get_enrollment(user, course_run_key)
+
+
+def get_phone_number(user_id):
+    try:
+        student = UserProfile.objects.get(user_id=user_id)
+        return student.phone_number
+    except UserProfile.phone_number.DoesNotExist as exception:
```

Comment:
I'm not sure what's happening here. You're attempting get a `UserProfile` via `user_id` in the try block above. If that `user_id` doesn't match a `UserProfile`, you're going to get a `UserProfile.DoesNotExist` exception. If you also want to check if the `phone_number` exists, that would be a separate check. 

Also, by raising the exception, we're commiting to causing a failure in the code that using this instead of handling it and logging it. 

I think we'd want something more in line with:
```
try:
    student = ...
except UserProfile.DoesNotExist:
    # log exception
    return None
return student.phone_number or None # Turns empty strings into None so we only have two return type, a phone number or None (which implies the user didn't exist or doesn't have a phone number). 
```

Reply:
changed the shape of the function to match the example.